pkg/odo/cli/component/deploy.go

```diff
@@ -173,11 +174,10 @@ func NewCmdDeploy(name, fullName string) *cobra.Command {
 	genericclioptions.AddContextFlag(deployCmd, &do.componentContext)
 
 	// enable devfile flag if experimental mode is enabled
-	if experimental.IsExperimentalModeEnabled() {
-		deployCmd.Flags().StringVar(&do.DevfilePath, "devfile", "./devfile.yaml", "Path to a devfile.yaml")
-		deployCmd.Flags().StringVar(&do.tag, "tag", "", "Tag used to build the image")
-		deployCmd.Flags().BoolVar(&do.deployOnly, "deployOnly", false, "Do not build the application, only deploy it")
-	}
+	deployCmd.Flags().StringVar(&do.DevfilePath, "devfile", "./devfile.yaml", "Path to a devfile.yaml")
```

Comment:
Comment still references "if experimental mode is enabled" but that if has been removed from the code.